### DISCUSSION TITLE
Doubly nested model upsert

### DISCUSSION THREAD

**Discussion by jamezening** [prisma/prisma#12980] [Q&A] [answered]
Hi,

I am a backend newb and have been working with Prisma/MongoDB.

Is there a way to execute a transaction that has a doubly nested upsert?
Or perhaps, should the data be modeled in a better way than below?

Take for example urbandictionary, where a user can create a word definition, and each word can have many definitions.

```
model User {
  id                  String   @id @default(auto()) @map("_id") @db.ObjectId
  email               String   @unique
  name                String
  Words           Word[]
}

model Word {
  id               String   @id @default(auto()) @map("_id") @db.ObjectId
  word             String   @unique
  author           User     @relation(fields: [authorId], references: [id])
  authorId         String   @db.ObjectId

  Definitions      Definition[]
}

model Definition {
  id               String   @id @default(auto()) @map("_id") @db.ObjectId
  definition       String
  word         Word @relation(fields: [wordId], references: [id])
  wordId       String   @db.ObjectId
}
```

Thank you for any help.

**Comment by austincrim** [comment, marked as answer]
Hey @vanilla-james 👋 ,

Prisma lets you do [nested writes](https://www.prisma.io/docs/concepts/components/prisma-client/transactions#nested-writes) arbitrarily. In your case, it would look something like this:
```ts
let result = await prisma.user.create({
  data: {
    email: 'austin@hello.com',
    name: 'austin',
    Words: {
      create: {
        word: 'foo',
        Definitions: {
          create: {
            definition: 'a definition'
          }
        }
      }
    }
  }
})
```

Nested writes are also transactional by nature. Let me know if I can clarify anything!

**Reply by jamezening** [reply]
Thank you for the help!
Yeah, I was struggling a bit with the third nested creation, but figured it out based on the friendly prisma documentation 👏.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
